lingo-share/src/components/video/videochat.jsx

```diff
@@ -1,28 +1,44 @@
 import React, { useState, useCallback, useEffect } from "react";
 import { get_token } from "../../firebase";
+import { UserAuth } from "../../contexts/AuthContext";
+import { useNavigate } from "react-router-dom";
 import Video from "twilio-video";
 import Lobby from "./lobby";
 import Room from "./room";
 
-const VideoChat = () => {
+const VideoChat = ({callerID}) => {
   const [username, setUsername] = useState("");
   const [roomName, setRoomName] = useState("");
   const [room, setRoom] = useState(null);
   const [connecting, setConnecting] = useState(false);
+  const { user } = UserAuth();
+  const navigate = useNavigate();
 
-  const handleUsernameChange = useCallback((event) => {
-    setUsername(event.target.value);
+  const handleLogout = useCallback(() => {
+    setRoom((prevRoom) => {
+      if (prevRoom) {
+        prevRoom.localParticipant.tracks.forEach((trackPub) => {
+          trackPub.track.stop();
+        });
+        prevRoom.disconnect();
+      }
+      return null;
+    });
+    navigate("/dashboard");
   }, []);
 
-  const handleRoomNameChange = useCallback((event) => {
-    setRoomName(event.target.value);
-  }, []);
+  const getRoomName = (uid, callerID) => {
+    const concatenatedRoomName = uid + callerID;
+    return concatenatedRoomName.split('').sort().join(''); // room name will be the concatenation of the two user IDs sorted alphabetically
+  }
 
-  const handleSubmit = useCallback(
-    async (event) => {
-      event.preventDefault();
+  
+  useEffect(() => {
+    setUsername(user.uid);
+    setRoomName(getRoomName(user.uid+callerID));
+    const handleSubmit = async () => {
       setConnecting(true);
-      const result = await get_token({ identity: username, room: roomName });
+      const result = await get_token({ identity: user.uid, room: roomName }); // idk why username is still null at this line so I replaced with user.uid
```

Comment:
@jqiao26 I feel like it's some React thing I couldn't figure

Reply:
I think for identity we actually want to use their real name/display name instead of uid.